scripts/variantstore/wdl/extract/hail_gvs_import.py

```diff
@@ -0,0 +1,148 @@
+export PYSPARK_SUBMIT_ARGS='--driver-memory 16g --executor-memory 16g pyspark-shell'
+gcloud auth application-default login
+curl -sSL https://broad.io/install-gcs-connector | python3
+```
+
+* Running the GVS to Hail VDS conversion script:
+
+`python hail_gvs_import.py`
+
+"""
+
+
+from google.cloud import storage
+
+import argparse
+import re
+
+
+def generate_avro_args(bucket, blob_prefix, key):
+    """
+    Generate a list of the Avro arguments for the `hl.import_gvs` invocation for the specified key. The datatype should
+    match the parameters of the `hl.import_gvs` function:
+
+    * vets (list of lists, one outer list per GVS superpartition of 4000 samples max)
+    * refs (list of lists, one outer list per GVS superpartition of 4000 samples max)
+    * sample_mapping (list)
+    * site_filtering_data (list)
+    * vqsr_filtering_data (list)
+    * vqsr_tranche_data (list)
+    """
+
+    keyed_prefix = f"{blob_prefix}/{key}/"
+
+    def superpartitioned_handler(blob_name):
+        relative_path = blob_name[len(keyed_prefix):]
+        parts = relative_path.split('/')
+
+        index = int(parts[0].split('_')[-1]) - 1
+        if len(ret) == index:
+            ret.append([])
+        ret[index].append(f'gs://{bucket.name}/{blob_name}')
+
+    def regular_handler(blob_name):
+        ret.append(f'gs://{bucket.name}/{blob_name}')
+
+    superpartitioned_keys = {'vets', 'refs'}
+    entry_handler = superpartitioned_handler if key in superpartitioned_keys else regular_handler
+
+    ret = []
+
+    # `list_blobs` paginates under the covers, explicit pagination not required regardless of the number of Avro files.
+    # https://stackoverflow.com/a/43646557
+    count = 0
+    log_interval = 1000
+    for blob in bucket.list_blobs(prefix=keyed_prefix):
+        count = count + 1
+        if count % log_interval == 0:
+            print(f"Processed {count} {key} blobs...")
+
+        if not blob.name.endswith(".avro"):
+            continue
+        entry_handler(blob.name)
+
+    return ret
+
+
+def import_gvs(bucket, object_prefix, vds_output_path, tmp_dir):
+    import hail as hl
+
+    unslashed_tmp_dir = tmp_dir if not tmp_dir.endswith('/') else tmp_dir[:-1]
+    hl.init(tmp_dir=f'{unslashed_tmp_dir}/hail_tmp_general')
+
+    rg38 = hl.get_reference('GRCh38')
+    rg38.add_sequence('gs://hail-common/references/Homo_sapiens_assembly38.fasta.gz',
+                      'gs://hail-common/references/Homo_sapiens_assembly38.fasta.fai')
+
+    def args(key):
+        return generate_avro_args(bucket, object_prefix, key)
+
+    hl.import_gvs(
+        vets=args('vets'),
+        refs=args('refs'),
+        sample_mapping=args('sample_mapping'),
+        site_filtering_data=args('site_filtering_data'),
+        vqsr_filtering_data=args('vqsr_filtering_data'),
+        vqsr_tranche_data=args('vqsr_tranche_data'),
+        reference_genome=rg38,
+        final_path=vds_output_path,
+        tmp_dir=f'{unslashed_tmp_dir}/hail_tmp_import_gvs'
```

Comment:
this is where we will put:
 intermediate_resume_point=x

and also do we want to add: 
 partitions_per_sample=0.4,